src/main/java/gov/cabinetofice/gapuserservice/service/jwt/impl/CustomJwtServiceImpl.java

```diff
@@ -147,4 +160,22 @@ public JwtPayload decodeTheTokenPayloadInAReadableFormat(DecodedJWT jwt) {
                 .email(email)
                 .build();
     }
+
+    public User getUserFromJwt(final HttpServletRequest request) {
+        final Cookie customJWTCookie = WebUtils.getCookie(request, userServiceCookieName);
+        if (customJWTCookie == null || customJWTCookie.getValue() == null) {
+            throw new UnauthorizedException("No JWT token provided");
+        }
+        final DecodedJWT decodedJwt = decodedJwt(customJWTCookie.getValue());
+        final JwtPayload payload = decodeTheTokenPayloadInAReadableFormat(decodedJwt);
+        return  userRepository.findBySub(payload.getSub()).get();
```

Comment:
userRepository.findBySub returns an optional so should check if it's present before getting it

Reply:
![Screenshot 2023-09-12 at 09 56 03](https://github.com/cabinetoffice/gap-user-service/assets/135221918/9a8e2903-206b-446e-b49c-6167d3f5ad48)

This method still has no coverage?

Reply:
![Screenshot 2023-09-12 at 11 39 29](https://github.com/cabinetoffice/gap-user-service/assets/135221918/f629088b-be1d-48d7-810d-c187487cd2aa)

Still no coverage for this method?

Reply:
pushed that now